cluster-autoscaler/go.mod

```diff
@@ -10,7 +10,7 @@ require (
 	github.com/Azure/go-autorest/autorest/azure/auth v0.5.8
 	github.com/Azure/go-autorest/autorest/date v0.3.0
 	github.com/Azure/go-autorest/autorest/to v0.4.0
-	github.com/aws/aws-sdk-go v1.38.49
+	github.com/aws/aws-sdk-go v1.41.3
```

Comment:
Do we need to update the entire SDK, or can we get away with a replace directive just for the `service/autoscaling` sub path, as this is pulling in far more in terms of new code than I would expect to be necessary (e.g. the entire ECR API.)

Reply:
I tried to follow the same approach eks did, but neither a `go mod tidy` nor a `go mod vendor` does anything and it seems subfolder replacements aren't really supported https://github.com/golang/go/issues/30886

Do you have an insights / guide on this could be done?

Reply:
I think you're correct, and the only other way we'd be able to achieve this is the path taken by some of the newer cloud providers to avoid adding new dependencies via vendoring under sub paths, e.g.: https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/exoscale/internal

@MaciekPytel do you have a concern about falling out of sync with the upstream dependency version here? Unfortunately it seems the upstream is still not at a version of the aws-sdk-go which supports this new functionality we're making use of.

Reply:
Ping :-)

Reply:
@MaciekPytel do you have any concerns here?

Reply:
This is exciting, rebased this PR :-)